content/quickstart/cpp/metrics.md

```diff
@@ -411,117 +411,136 @@ int main(int argc, char **argv) {
 #include "opencensus/exporters/stats/prometheus/prometheus_exporter.h"
+#include "opencensus/stats/stats.h"
+#include "opencensus/tags/tag_key.h"
 #include "prometheus/exposer.h"
 
-ABSL_CONST_INIT const absl::string_view kLatencyMeasureName     = "repl/latency";
-ABSL_CONST_INIT const absl::string_view kLatencyViewName        = "ocquickstart.io/latency";
-ABSL_CONST_INIT const absl::string_view kLinesCountViewName     = "ocquickstart.io/lines_in";
-ABSL_CONST_INIT const absl::string_view kLineLengthsMeasureName = "repl/line_lengths";
-ABSL_CONST_INIT const absl::string_view kLineLengthsViewName    = "ocquickstart.io/line_lengths";
-
-static const opencensus::stats::MeasureDouble m_latency_ms =
-                    opencensus::stats::MeasureDouble::Register(
-                            kLatencyMeasureName, "The latency in milliseconds", "ms");
+namespace {
 
-static const opencensus::stats::MeasureInt64 m_line_lengths =
-                    opencensus::stats::MeasureInt64::Register(
-                            kLineLengthsMeasureName, "The distributions of line lengths", "By");
+ABSL_CONST_INIT const absl::string_view kLatencyMeasureName = "repl/latency";
+ABSL_CONST_INIT const absl::string_view kLineLengthsMeasureName =
+    "repl/line_lengths";
 
-static const opencensus::tags::TagKey key_method =
-                    opencensus::tags::TagKey::Register("method");
+// Measures and TagKeys are not POD. Treat them as singletons and initialize on
+// demand in order to avoid initialization order issues.
 
-std::string processLine(std::string in) {
-    absl::Time start = absl::Now();
-    std::string out(in);
-
-    for (auto it = out.begin(); it != out.end(); it++)
-        *it = std::toupper(*it);
-
-    absl::Time end = absl::Now();
-    double latency_ms = absl::ToDoubleMilliseconds(end - start);
-
-    opencensus::stats::Record({{m_latency_ms, latency_ms}}, {{key_method, "processLine"}});
-
-    return out;
+opencensus::stats::MeasureDouble LatencyMsMeasure() {
+  static const auto measure = opencensus::stats::MeasureDouble::Register(
+      kLatencyMeasureName, "The latency in milliseconds", "ms");
+  return measure;
 }
 
-void registerAsView(opencensus::stats::ViewDescriptor vd) {
-    opencensus::stats::View view(vd);
-    vd.RegisterForExport();
+opencensus::stats::MeasureInt64 LineLengthsMeasure() {
+  static const auto measure = opencensus::stats::MeasureInt64::Register(
+      kLineLengthsMeasureName, "The distributions of line lengths", "By");
+  return measure;
 }
 
-std::string getLine() {
-    absl::Time start = absl::Now();
+opencensus::tags::TagKey MethodKey() {
+  static const auto key = opencensus::tags::TagKey::Register("method");
+  return key;
+}
 
-    std::string input;
+std::string processLine(const std::string& in) {
+  absl::Time start = absl::Now();
+  std::string out(in);
 
-    // Get the line
-    std::getline(std::cin, input);
+  for (auto it = out.begin(); it != out.end(); it++) *it = std::toupper(*it);
```

Comment:
In the future, perhaps let's make this more readable by putting the assignment on a newline
```cpp
for (auto it = out.begin(); it != out.end(); it++)
      *it = std::toupper(*it);
```

Reply:
clang-format strikes again. Maybe add braces.